keyboards/model_f_labs/xwhatsit_util_comm.c

```diff
@@ -0,0 +1,228 @@
+/* Copyright 2020 Purdea Andrei
+ * Copyright 2021-2022 Matthew J Wolf
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#include "raw_hid.h"
+#include "xwhatsit_util_comm.h"
+#include "xwhatsit_capsense_matrix.h"
+#include <string.h>
+#include <platforms/eeprom.h>
+
+extern const char *KEYBOARD_FILENAME; // This must be defined in keyboard_name.c to equal the filename. This is sent back to the PC-side software for it to determine which keyboard we are using.
```

Comment:
This is an item I can not easily change. The utility that uses this code supports many keyboards. Changing this would affect all the other keyboards.